CRVResponse/src/CrvPlot_module.cc

```diff
@@ -176,129 +179,165 @@ namespace mu2e
       hist->GetYaxis()->SetLabelColor(kBlue);
       hist->Draw();
 
-      double histMax = hist->GetMaximum();
-      if(histMax==0) histMax=1;
+      if(histMax[SiPM]<hist->GetMaximum()) histMax[SiPM]=hist->GetMaximum();
+    }
 
 //SiPM response
-      std::vector<double> siPMtimes, siPMcharges;
-      if(siPMCharges!=crvSiPMChargesCollection->end())
+    std::vector<double> siPMtimes[4], siPMcharges[4];
+    CrvSiPMChargesCollection::const_iterator charges;
+    for(charges=crvSiPMChargesCollection->begin(); charges!=crvSiPMChargesCollection->end(); charges++)
+    {
+      if(charges->GetScintillatorBarIndex()==barIndex)
       {
-        const std::vector<CrvSiPMCharges::CrvSingleCharge> &timesAndCharges = siPMCharges->second.GetSiPMCharges(SiPM);
+        int SiPM=charges->GetSiPMNumber();
+        if(SiPM<0 || SiPM>3) continue;
+
+        const std::vector<CrvSiPMCharges::SingleCharge> &timesAndCharges = charges->GetCharges();
         for(size_t i=0; i<timesAndCharges.size(); i++)
         {
-          siPMtimes.push_back(timesAndCharges[i]._time);
-          siPMcharges.push_back(timesAndCharges[i]._chargeInPEs);
+          siPMtimes[SiPM].push_back(timesAndCharges[i]._time);
+          siPMcharges[SiPM].push_back(timesAndCharges[i]._chargeInPEs);
         }
       }
+    }
+
+    for(int SiPM=0; SiPM<4; SiPM++)
+    {
+      _canvas[_icanvas]->cd(SiPM+1);
 
       double scaleSiPMResponse = 0.5;
-      TH1D *histSiPMResponse=new TH1D((s2.str()+"SiPMResponse").c_str(),"",100,_timeStart,_timeEnd+extension);
+      std::ostringstream s4;
+      s4<<"SiPMcharges_"<<_icanvas<<"__"<<SiPM;
+      TH1D *histSiPMResponse=new TH1D(s4.str().c_str(),"",100,_timeStart,_timeEnd);
       histVector.push_back(histSiPMResponse);
-      for(unsigned int i=0; i<siPMtimes.size(); i++)
+      for(unsigned int i=0; i<siPMtimes[SiPM].size(); i++)
       {
-        histSiPMResponse->Fill(siPMtimes[i], siPMcharges[i]*scaleSiPMResponse);
+        histSiPMResponse->Fill(siPMtimes[SiPM][i], siPMcharges[SiPM][i]*scaleSiPMResponse);
       }
       histSiPMResponse->SetLineColor(kOrange);
       histSiPMResponse->Draw("same");
 
+      if(histMax[SiPM]==0) histMax[SiPM]=1;
+
+      TGaxis *axisSiPMResponse = new TGaxis(_timeEnd,0,_timeEnd,histMax[SiPM],0,histMax[SiPM]/scaleSiPMResponse,10,"+L");
+      axisVector.push_back(axisSiPMResponse);
+      axisSiPMResponse->SetTitle("SiPM charges [PE]");
+      axisSiPMResponse->SetTitleOffset(1.0);
+      axisSiPMResponse->SetTitleColor(kOrange);
+      axisSiPMResponse->SetLineColor(kOrange);
+      axisSiPMResponse->SetLabelColor(kOrange);
+      axisSiPMResponse->Draw("same");
+    }
+
 //waveforms
-      double scale=NAN;
-      for(size_t digiIndex=0; digiIndex<crvDigiCollection->size(); digiIndex++)
+    std::vector<std::vector<double> > ADCs[4];  //there can be multiple disconnected ADC waveforms per SiPM
+    std::vector<std::vector<double> > times[4];
+    double scale[4]={NAN};
+    double maxADC[4]={NAN};
```

Comment:
using NAN as a flag risks non-sensical downstream behavior and makes the code hard to interpret.  Consider using FLTMAX or FLTMIN or a bool flag (initialized) instead.

Reply:
This module is only used for plotting photon times, waveforms, etc. and does not produce any data.